src/DFG.cpp

```diff
@@ -49,6 +38,75 @@ DFG::DFG(Function& t_F, list<Loop*>* t_loops, bool t_targetFunction,
 
 }
 
+// Specilized fusion for the nonlinear operations.
+void DFG::nonlinear_combine() {
+  tuneForBitcast();
+  combineMulAdd(2);
+  combinePhiAdd(1);
+  combine("fcmp", "select", 1);
+  combine("icmp", "select", 1);
+  combine("icmp", "br", 2);
+  combine("fcmp", "br", 2);
+  combineAddAdd(1);
+  tuneForPattern();
+  tuneDivPattern();
+}
+
+// For division, we regrad it as non-vectorized instructions, which is contradictory to LLVM Pass.
+// Thus we need to split a vectorization divison into multiple scalar divisions.
```

Comment:
If it is an integer divide, we still can treat it as vectorized div as the HW cost is affordable? Should we provide a user option to enable this tuning?

Reply:
I agree with the statement. So you mean I added a new option in the `param.json` to achieve it?

Maybe for an integer divide, we can pass the vectorizaion factor 1 to the mapper, then it will not split division node and the DFG is also the same.